src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -0,0 +1,433 @@
+  u_boundary       ::Array{uEltype, NDIMST2M1} # NDIMS * 2 - 1
+  other_semi_index ::Int
+  other_orientation::Int
+  indices          ::Indices
+
+  function BoundaryConditionCoupled(other_semi_index, indices, uEltype)
+    NDIMS = length(indices)
+    u_boundary = Array{uEltype, NDIMS*2-1}(undef, ntuple(_ -> 0, NDIMS*2-1))
+
+    if indices[1] in (:begin, :end)
+      other_orientation = 1
+    elseif indices[2] in (:begin, :end)
+      other_orientation = 2
+    else # indices[3] in (:begin, :end)
+      other_orientation = 3
+    end
+
+    new{NDIMS, NDIMS*2-1, uEltype, typeof(indices)}(
+      u_boundary, other_semi_index, other_orientation, indices)
+  end
+end
+
+Base.eltype(boundary_condition::BoundaryConditionCoupled) = eltype(boundary_condition.u_boundary)
+
+function (boundary_condition::BoundaryConditionCoupled)(u_inner, orientation, direction,
+                                                        cell_indices, surface_node_indices,
+                                                        surface_flux_function, equations)
+  # get_node_vars(boundary_condition.u_boundary, equations, solver, surface_node_indices..., cell_indices...),
+  # but we don't have a solver here
+  u_boundary = SVector(ntuple(v -> boundary_condition.u_boundary[v, surface_node_indices..., cell_indices...],
+                              Val(nvariables(equations))))
+
+  # Calculate boundary flux
+  if iseven(direction) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
+end
+
+
+function allocate_coupled_boundary_conditions(semi::AbstractSemidiscretization)
+  n_boundaries = 2 * ndims(semi)
+  mesh, equations, solver, _ = mesh_equations_solver_cache(semi)
+
+  for direction in 1:n_boundaries
+    boundary_condition = semi.boundary_conditions[direction]
+
+    allocate_coupled_boundary_condition(boundary_condition, direction, mesh, equations, solver)
+  end
+end
+
+
+# Don't do anything for other BCs than BoundaryConditionCoupled
+function allocate_coupled_boundary_condition(boundary_condition, direction, mesh, equations, solver)
+end
+
+# In 2D
+function allocate_coupled_boundary_condition(boundary_condition::BoundaryConditionCoupled{2},
+                                             direction, mesh, equations, dg::DGSEM)
+  if direction in (1, 2)
+    cell_size = size(mesh, 2)
+  else
+    cell_size = size(mesh, 1)
+  end
+
+  uEltype = eltype(boundary_condition)
+  boundary_condition.u_boundary = Array{uEltype, 3}(undef, nvariables(equations), nnodes(dg),
+                                                    cell_size)
+end
+
+# Don't do anything for other BCs than BoundaryConditionCoupled
+function copy_to_coupled_boundary!(boundary_condition, u_ode, semi) end
```

Comment:
```suggestion
# Don't do anything for other BCs than BoundaryConditionCoupled
function copy_to_coupled_boundary!(boundary_condition, u_ode, semi)
  return nothing
end
```
This clarifies that you're really doing nothing on purpose instead of trying to write a docstring or just creating a new function without methods in the wrong way. We had a bug like this a few times, so it would be nice to clarify that this is not the case here.

Reply:
Fixed in 476d1b30e.